src/ibusengine.c

```diff
@@ -303,6 +305,7 @@ static const gchar introspection_xml[] =
     /* FIXME properties */
     "    <property name='ContentType' type='(uu)' access='write' />"
     "    <property name='FocusId' type='(b)' access='read' />"
+    "    <property name='NeedsSurroundingText' type='(b)' access='read' />"
```

Comment:
I'm thinking about D-Bus property names might be  a noun since it can be modifed with get/set D-Bus methods.
So "get/set a needs surrounding text" might be strange for me. Also the meaning of NeedsSurroundingText is not clarified because ibus_engine_get_surrounding_text() works even if the property is FALSE.
"ActiveSurroundingText" might be good and all internal variables are replaced has_active_surrounding_text with needs_surrounding_text.